src/diffusers/pipelines/auto_pipeline.py

```diff
@@ -305,8 +308,6 @@ def from_pretrained(cls, pretrained_model_or_path, **kwargs):
         use_auth_token = kwargs.pop("use_auth_token", None)
         local_files_only = kwargs.pop("local_files_only", False)
         revision = kwargs.pop("revision", None)
-        subfolder = kwargs.pop("subfolder", None)
```

Comment:
Ideally this should have been fixed in another PR, but doing it here now instead. `DiffusionPipeline` never consumes neither `subfolder` nor `user_agent` so we shouldn't add it to the Auto... pipelines cc @yiyixuxu just FYI